lib/inspec/profile.rb

```diff
@@ -709,7 +709,8 @@ def check # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metri
     def validity_check # rubocop:disable Metrics/AbcSize
       errors = []
       warnings = []
-      info = info_from_parse
+      yml_metadata = Metadata.from_file(@source_reader.target.abs_path(@source_reader.metadata.ref), nil) || {}
```

Comment:
Is this really needed? Can't you just call self.metadata.params here? Why are we duplicating the logic about how to locate and load a metadata file in the middle of the check logic?